cpp/daal/src/algorithms/dtrees/forest/regression/df_regression_train_dense_default_impl.i

```diff
@@ -144,10 +145,13 @@ public:
         DAAL_ASSERT(n > 0);
         node.response = imp.mean;
 #ifdef DEBUG_CHECK_IMPURITY
-        algorithmFPType response;
-        algorithmFPType val = calcResponse(response, idx, n);
-        node.response       = response;
-        DAAL_ASSERT(fabs(val - imp.mean) < 0.001);
+        if (!this->_weight)
+        {
+            algorithmFPType response;
+            algorithmFPType val = calcResponse(response, idx, n);
+            node.response       = response;
+            DAAL_ASSERT(fabs(val - imp.mean) < 0.001);
```

Comment:
Do we need to change threshold since weights support was added?

Reply:
I suggest leaving it as it is

Reply:
Why?